### ISSUE TITLE
Reimplement core::uuid_generate() using libuuid

### ISSUE THREAD

**Issue by gavv** [roc-streaming/roc-toolkit#692]
We have [simple implementation](https://github.com/roc-streaming/roc-toolkit/blob/develop/src/internal_modules/roc_core/uuid.cpp) of UUID generation using random numbers. It would be great to have alternative optional implementation using [**libuuid**](https://man7.org/linux/man-pages/man3/uuid_generate.3.html) from [util-linux](https://github.com/util-linux/util-linux/tree/master/libuuid) package.

Our implementation is portable but unsophisticated. It blindly generates random UUID no matter where it's running and what's the quality of PRNG. libuuid, on the other hand, is capable to check system capabilities: if high-quality random generator is available, it will use it, otherwise it will fallback to using low-level system information like MAC address and system time. It will also employ uuidd if it's present on system. 

Steps:

* Add libuuid to build scripts ([build-3rdparty.py](https://github.com/roc-streaming/roc-toolkit/blob/develop/scripts/scons_helpers/build-3rdparty.py), [3rdparty/Sconscript](https://github.com/roc-streaming/roc-toolkit/blob/develop/3rdparty/SConscript)). Add `--disable-libuuid` to [SConstruct](https://github.com/roc-streaming/roc-toolkit/blob/develop/SConstruct). If the option is given, enable `target_nouuid` [target directory](https://roc-streaming.org/toolkit/docs/internals/code_structure.html#targets), otherwise enable `target_libuuid`.

   Basically you can replicate what have been done in [this commit](https://github.com/roc-streaming/roc-toolkit/commit/808b126f111a5ce376831cac583dfb80fb9b97dc) to add OpenSSL (you can ignore `openssl_get_platform()` function, it's OpenSSL-specific).

* Move existing implementation from `roc_core` to `roc_core/target_nouuid` target directory.

* Add new implementation to `roc_core/target_libuuid` using libuuid.

* Add simple test that will pass for both implementations.

Related docs:

* https://roc-streaming.org/toolkit/docs/development/continuous_integration.html
* https://roc-streaming.org/toolkit/docs/building/developer_cookbook.html

**Comment by nolan-veed** [comment]
This one is right up my street. I'll do it.

**Comment by gavv** [comment]
Landed
